### DISCUSSION TITLE
Shorthand opacity does not work with CSS varvariables

### DISCUSSION THREAD

**Discussion by jLynx** [tailwindlabs/tailwindcss#12622] [Help] [answered]
I am encountering an issue when utilizing CSS variables with TailwindCSS. Allow me to provide some context and examples below:

When I have Tailwind configured to consume a CSS variable as follows:

```javascript
module.exports = {
  theme: {
    extend: {
      colors: {
        backgroundBase: "var(--backgroundBase)"
      }
    }
  }
}
```
I run into an inconsistency with its expected behaviour. For instance, when I declare in my HTML:

```html
<p className="bg-backgroundBase/80">
```
The issue here is that the background colour does not apply at all which also means no opacity. But if I do:
```html
<p className="bg-backgroundBase bg-opacity-80">
```
it works as expected, However, I am getting a warning saying `Classname 'bg-opacity-80' should be replaced by an opacity suffix (eg. '/80')`
Yet when I tried that as I stated above, it does not do anything.

However, if I remove the CSS variable and just hardcode the `tailwind.config.ts` as such:

```javascript
module.exports = {
  theme: {
    extend: {
      colors: {
        backgroundBase: "#FFFFFF"
      }
    }
  }
}
```

then the value, then `bg-backgroundBase/80` works perfectly fine. The issue is I need to use CSS variables for all of my tailwind config for updating my theme dynamically.

As as far as I know doing this is no longer possible when using css vaiables
`from-gradientPrimarySubtlest/20 to-gradientTertiaryOneSubtlest/80 bg-gradient-to-r to-50%` as the opacity is different from one side of the gradient to the other

Does anyone know a method to enable TailwindCSS to integrate with shorthand opacity when CSS variables are involved? It would be greatly appreciated!

**Comment by fordaaron** [comment, marked as answer]
See https://tailwindcss.com/docs/customizing-colors#using-css-variables

**Reply by jLynx** [reply]
Sadly doesnt answer my question, as I want the base colour the be solid, but also want to be able to use that base colour with opacity such as `<div className="bg-backgroundBase/80">div with opacity</div>` and `<div className="bg-backgroundBase">div without opacity</div>`

Which works fine with
```
module.exports = {
  theme: {
    extend: {
      colors: {
        backgroundBase: "#FFFFFF"
      }
    }
  }
}
```

But not with

```
module.exports = {
  theme: {
    extend: {
      colors: {
        backgroundBase: "var(--backgroundBase)"
      }
    }
  }
}
```

**Reply by jLynx** [reply]
Scratch that, this worked. I was just returning `var(--backgroundBase)` instead of `rgb(var(--backgroundBase))`

**Comment by crswll** [comment]
You can use `color-mix` to make this a lot easier. It will also keeps your color in the format you want.

```
colors: {
  primary: "color-mix(in srgb, var(--colors-primary), transparent calc(100% - 100% * <alpha-value>))",
}
```

Here's a little demo: https://play.tailwindcss.com/4zw0y2U6wT?file=config
